packages/jupyter-chat/src/components/input/buttons/send-button.tsx

```diff
@@ -100,19 +100,22 @@ export function SendButton(
 
   function sendWithSelection() {
     let source = '';
-
+    let language: string | undefined;
     if (selectionWatcher?.selection) {
       // Append the selected text if exists.
       source = selectionWatcher.selection.text;
+      language = selectionWatcher.selection.language;
     } else if (activeCellManager?.available) {
       // Append the active cell content if exists.
-      source = activeCellManager.getContent(false)!.source;
+      const content = activeCellManager.getContent(false);
+      source = content!.source;
+      language = content?.language;
     }
     let content = model.value;
     if (source) {
       content += `
 
-\`\`\`
+\`\`\`${language ?? ''}
```

Comment:
There are potentially 3 different lists of languages:

1. From file editor models: `(await languages.getLanguage(editor?.model.mimeType))?.name;`
2. From notebook models: `this._notebookTracker.currentWidget?.model?.defaultKernelLanguage`
3. The list of language identifiers recognized by the Rendermime plugin that renders these code blocks
 
We should try to find source code references for all 3. If this lists do not exactly match, that means there are edge cases in which this does not work. We don't need to handle them all, but we should probably open an issue that includes all 3 lists for future developers to fix these edge cases.